### ISSUE TITLE
`BufferedReader.read_line` undocumented behavior on new line characters

### ISSUE THREAD

**Issue by Coachonko** [vlang/v#18751]
### Describe the issue

The documentation says the reader stops at `\n` but it does not do that: replies from a Redis server always end in `\r\n` and `read_reply` does not return these characters.
This is a problem for me, what alternatives do I have to read the whole line including the `\r\n`?

### Links

Problem was identified [here](https://github.com/Coachonko/redis/blob/6150da3e2b293af90dca9eb5a341a0eb7dea4eff/src/pool/proto/reader.v#L49)
[read_line](https://github.com/vlang/v/blob/master/vlib/io/buffered_reader.v#L111)

**Comment by JalonSolov** [comment]
Yes, the documentation should say that it reads until EOL, and returns the string without it.

`read_line` is not undocumented, simply slightly incorrectly documented.

If you want different behaviour, call `BufferedReader.read` instead of `BufferedReader.read_line`.

**Comment by Casper64** [comment]
Maybe I misunderstand the issue, but `read_line` does also stop at `\r\n` it's in the code:

```
// do some checking for whether we hit \r\n or just \n
```

It will actually check if there was a `\r` first and then returns the line without `\r\n`.

Or do you need to check whether the line ended in `\r\n`?

**Comment by Coachonko** [comment]
> Or do you need to check whether the line ended in `\r\n`?

Yes I expected read_line to return the line including `\n` and the characters before it, since it is described as "stopping" there.

I have solved the problem by providing my own buffer and reading implementation on TcpConn.reader (since I am reading a TcpConn stream). Proper documentation would have saved me a lot of time I spent debugging the situation.

**Comment by JalonSolov** [comment]
All of the `read_line`/`read_lines` routines in vlib return the line _without_ the EOL.  I don't see this explicitly documented, so the documentation could be improved.

It's done this way because that's what 99.9% of programs expect... the line without the EOL.

And you could still use `read` instead of `read_line`.
